src/Features/Core/Portable/EditAndContinue/Remote/IRemoteEditAndContinueService.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.EditAndContinue
+{
+    internal interface IRemoteEditAndContinueService
+    {
+        internal interface IStartEditSessionCallback
+        {
+            Task<ImmutableArray<ActiveStatementDebugInfo>> GetActiveStatementsAsync(CancellationToken cancellationToken);
+            Task<(int errorCode, string? errorMessage)?> GetEncAvailabilityAsync(Guid mvid, CancellationToken cancellationToken);
+            Task PrepareModuleForUpdateAsync(Guid mvid, CancellationToken cancellationToken);
+        }
+
+        internal interface IDocumentActiveStatementProviderCallback
+        {
+            Task<ImmutableArray<TextSpan>> GetSpansAsync(CancellationToken cancellationToken);
+        }
+
+        internal interface ISolutionActiveStatementProviderCallback
+        {
+            Task<ImmutableArray<TextSpan>> GetSpansAsync(DocumentId documentId, CancellationToken cancellationToken);
+        }
```

Comment:
for my own understanding, what's teh difference between a solution-active-statement and a document-active-statement.  i get the latter... i don't really understand the former.

Reply:
IDocumentActiveStatementProviderCallback instance captures a specific document instance while ISolutionActiveStatementProviderCallback captures solution instance and needs documentId to identify which document to return spans for.